src/nvim/window.c

```diff
@@ -490,7 +487,8 @@ void do_window(int nchar, long Prenum, int xchar)
 wingotofile:
     CHECK_CMDWIN;
 
-    ptr = (char *)grab_file_name(Prenum1, &lnum);
+    linenr_T lnum = -1;
+    char *ptr = (char *)grab_file_name(Prenum1, &lnum);
```

Comment:
Moving the declaration of `ptr` here looks a bit strange.

Reply:
Didn't realize this was in scope for other cases.

I've surrounded in `{}` and added `ptr` decls for the other cases.